system/src/system_network_manager.cpp

```diff
@@ -800,6 +800,28 @@ NetworkManager::InterfaceRuntimeState* NetworkManager::getInterfaceRuntimeState(
     return nullptr;
 }
 
+bool NetworkManager::isInterfacePowerState(if_t iface, if_power_state_t state) const {
+    bool ret = false;
+    if_power_state_t pwr = IF_POWER_STATE_NONE;
+    if (!iface) {
+        for_each_iface([&](if_t iface, unsigned int curFlags) {
+            if (if_get_power_state(iface, &pwr) != SYSTEM_ERROR_NONE) {
+                return;
```

Comment:
Would this just return the first network interface found when no specific network IF type is specified?  E.g., Network.isOn() could return on/off when the first IF is found?  I'm not sure if that's good or bad, but some comments added to this function to explain it's behavior would be helpful.

Also, I don't think this should be returning true/false (which is missing) but likely doing something with the following `if (pwr == state) { }`.

Reply:
It doesn't take into account just the first interface. Lambda is executed for each of them. If at least one is on - return true.

Reply:
Here we just return the lambda function with `void`, not returning the `isInterfacePowerState()`.

Reply:
I would still love to see some comments added to this function to describe its behavior.  Something like, "This function checks each network interface for the intended power state, and returns true when the first one that matches the desired state is found.  Some remaining network interfaces might not be checked."

Reply:
Will do.

Reply:
>Some remaining network interfaces might not be checked

If you go into the `for_each_iface()` you'll see that every interface is checked. The `return` here only exits the lambda function. I'll still add the comment, but it would be like:
```
This function checks each network interface for the intended power state, and returns true as long as one that matches the desired state is found
```